shared/serial_port_controller.py

```diff
@@ -28,12 +28,18 @@ def __init__(self, setting_file = None):
         self.reader_port = None
 
         if setting_file:
-            file_path = os.getcwd() + "\\settings\\serial ports\\preference\\" + setting_file
+            if os.name == 'posix':
```

Comment:
All of these OS detection changes are relatively the same, minus what arguments are being passed into the file explorer on the respective system. They are all structured correctly and look good to me.